alpaca_trade_api/stream2.py

```diff
@@ -98,12 +100,44 @@ def __init__(self, key_id=None, secret_key=None, base_url=None):
             await self._ensure_nats()
             await self.polygon.subscribe(nats_channels)
 
+    async def test(self, poll_delay=1, max_buffer_length=1000, reconnect_delay=10, initial_channels=[]):
+        buffer = io.StringIO()
+        logger = logging.getLogger('websockets')
+        logger.setLevel(logging.DEBUG)
+        logger.addHandler(logging.StreamHandler(stream=buffer))
+
+        while True:
+            await asyncio.sleep(poll_delay)
+            buffer_text = buffer.getvalue()
+            if 'CLOSED' in buffer_text:
+                print('stream closed')
+                while True:
+                    try:
+                        print('attempting to reconnect...', end='\r')
+                        # self._ws = await self._connect()
```

Comment:
This line can be deleted, right?

Reply:
Yep, that was my original attempt that didn't pan out. It's not enough to reconnect, we also need to re-subscribe.